app/models/metric/ci_mixin/state_finders.rb

```diff
@@ -48,14 +48,25 @@ def vim_performance_state_association(ts, assoc)
       @last_vps_ts = ts
       # we are using a different timestamp
       # clear out relevant associations
-      (VimPerformanceState::ASSOCIATIONS & self.class.reflections.keys.map(&:to_sym)).each do |vps_assoc|
-        association(vps_assoc).reset
+      vim_performance_state_child_associations.each do |_vps_association, model_association|
+        association(model_association).reset
       end
     end
 
+    # TODO: do we need to translate vms to proper name / use child_associations hash?
+    # specifically thinking about vms
     if !association(assoc.to_sym).loaded?
       MiqPreloader.preload_from_array(self, assoc, vim_performance_state_for_ts(ts).public_send(assoc))
     end
     public_send(assoc)
   end
+
+  # @returns Hash<name in VimPerformanceState, (virtual?) association name in model>
+  def vim_performance_state_child_associations
+    if defined?(self.class::PERF_ROLLUPS)
+      self.class::PERF_ROLLUPS
+    else
+      self.class::PERF_ROLLUP_CHILDREN.each_with_object({}) { |x, h| h[x] = x }
+    end
```

Comment:
This makes me think that instead of introducing a new constant that partially duplicates PERF_ROLLUP_CHILDREN that we should just outright change PERF_ROLLUP_CHILDREN.

Reply:
I understand that they serve 2 different purposes, so perhaps the other option is a better name than PERF_ROLLUPS; perhaps VIM_PERFORMANCE_STATE_ROLLUP_CHILDREN?